pkg/scheduling/volumeusage.go

```
@@ -77,39 +80,29 @@ func (u Volumes) Insert(volumes Volumes) {
 func GetVolumes(ctx context.Context, kubeClient client.Client, pod *v1.Pod) (Volumes, error) {
 	ctx = logging.WithLogger(ctx, logging.FromContext(ctx).With("pod", pod.Name))
 	podPVCs := Volumes{}
-	defaultStorageClassName, err := DiscoverDefaultStorageClassName(ctx, kubeClient)
-	if err != nil {
-		return nil, fmt.Errorf("discovering default storage class, %w", err)
-	}
 	for _, volume := range pod.Spec.Volumes {
 		ctx = logging.WithLogger(ctx, logging.FromContext(ctx).With("volume", volume.Name))
-		var pvcID, storageClassName, volumeName string
-		var pvc v1.PersistentVolumeClaim
-		if volume.PersistentVolumeClaim != nil {
-			if err = kubeClient.Get(ctx, client.ObjectKey{Namespace: pod.Namespace, Name: volume.PersistentVolumeClaim.ClaimName}, &pvc); err != nil {
-				return nil, err
-			}
-			pvcID = fmt.Sprintf("%s/%s", pod.Namespace, volume.PersistentVolumeClaim.ClaimName)
-			storageClassName = lo.FromPtr(pvc.Spec.StorageClassName)
-			volumeName = pvc.Spec.VolumeName
-		} else if volume.Ephemeral != nil {
-			// generated name per https://kubernetes.io/docs/concepts/storage/ephemeral-volumes/#persistentvolumeclaim-naming
-			pvcID = fmt.Sprintf("%s/%s-%s", pod.Namespace, pod.Name, volume.Name)
-			storageClassName = lo.FromPtr(volume.Ephemeral.VolumeClaimTemplate.Spec.StorageClassName)
-			volumeName = volume.Ephemeral.VolumeClaimTemplate.Spec.VolumeName
-		} else {
+		pvc, err := volumeutil.GetPersistentVolumeClaim(ctx, kubeClient, pod, volume)
+		if errors.IsNotFound(err) {
```

Why do we chose to ignore the error here? When we have handling logic like this we should be really explicit about why we are choosing to swallow the error.

---

I believe this was the case we discussed offline where for this to occur a finalizer would have to be removed from the PVC. In that case it won't naturally recover but the node will stay scheduled. This ensures we are able to make progress.  I can add a comment about that here since it's not clear.